### PR TITLE
Draft PR for API download button

### PR DESCRIPTION
**Pull request by VimukthiMayadunne**
Draft PR for API download button

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/utils/APIExportUtil.java [hunk 1]
```diff
@@ -893,4 +894,27 @@ private static void exportClientCertificates(String archivePath, API api, int te
...
+    /**
+     * Exports an API from API Manager for a given API ID. Meta information, API icon, documentation, WSDL
+     * and sequences are exported. This service generates a zipped archive which contains all the above mentioned
+     * resources for a given API.
+     *
+     * @param preserveStatus Preserve API status on export
+     * @return Zipped file containing exported API
+     */
+
+    public static File exprotApi (APIProvider apiProvider, APIIdentifier apiIdentifier, String userName, ExportFormat exportFormat, Boolean preserveStatus) {
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/utils/APIExportUtil.java]: exprotApi typo

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/utils/APIExportUtil.java [hunk 2]
```diff
@@ -893,4 +894,27 @@ private static void exportClientCertificates(String archivePath, API api, int te
...
+        API api;
+        APIImportExportManager apiImportExportManager;
+        boolean isStatusPreserved = preserveStatus == null || preserveStatus;
+        try {
+            api = apiProvider.getAPI(apiIdentifier);
+            apiImportExportManager = new APIImportExportManager(apiProvider, userName);
+            File files = apiImportExportManager.exportAPIArchive(api, isStatusPreserved, exportFormat);
+            return files;
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/utils/APIExportUtil.java]: return apiImportExportManager.exportAPIArchive(api, isStatusPreserved, exportFormat);

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/utils/APIExportUtil.java [hunk 3]
```diff
@@ -893,4 +894,27 @@ private static void exportClientCertificates(String archivePath, API api, int te
...
+        } catch (APIImportExportException | APIManagementException e) {
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/utils/APIExportUtil.java]: No need to catch those APIM errors. Throw from the method

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/impl/ExportApiServiceImpl.java [hunk 4]
```diff
@@ -71,58 +63,8 @@
...
-            File file = apiImportExportManager.exportAPIArchive(api, isStatusPreserved, exportFormat);
-            return Response.ok(file)
-                    .header(RestApiConstants.HEADER_CONTENT_DISPOSITION, "attachment; filename=\""
-                            + file.getName() + "\"")
-                    .build();
-        } catch (APIManagementException | APIImportExportException e) {
-            RestApiUtil.handleInternalServerError("Error while exporting " + RestApiConstants.RESOURCE_API, e, log);
-        }
-        return null;
+        ExportApiUtil exportApi=new ExportApiUtil();
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/impl/ExportApiServiceImpl.java]: format

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ExportApiServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+     * @param version        Version of the API that needs to be exported
+     * @param providerName   Provider name of the API that needs to be exported
+     * @param format         Format of output documents. Can be YAML or JSON
+     * @param preserveStatus Preserve API status on export
+     * @return Zipped file containing exported API
+     */
+
+    public Response exportApiGet(String name, String version, String providerName, String format, Boolean preserveStatus, MessageContext messageContext) throws APIManagementException {
+        ExportApiUtil exportApiUtil = new ExportApiUtil();
+        return exportApiUtil.exportApiGet(name, version, providerName, format, preserveStatus);
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ExportApiServiceImpl.java]: Use proper method name

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/impl/ExportApiUtil.java [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+     *
+     * @param name           Name of the API that needs to be exported
+     * @param version        Version of the API that needs to be exported
+     * @param providerName   Provider name of the API that needs to be exported
+     * @param format         Format of output documents. Can be YAML or JSON
+     * @param preserveStatus Preserve API status on export
+     * @return Zipped file containing exported API
+     */
+
+    public Response exportApiGet(String name, String version, String providerName, String format, Boolean preserveStatus) {
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/java/org/wso2/carbon/apimgt/rest/api/util/impl/ExportApiUtil.java]: use proper method name: exportAPI

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ExportApiServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
+package org.wso2.carbon.apimgt.rest.api.publisher.v1.impl;
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ExportApiServiceImpl.java]: License header

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml [hunk 8]
```diff
@@ -6400,6 +6400,90 @@ paths:
             Not Found.
             Requested role name does not exist.
 
+  ######################################################
+  # Export Resource API
+  ######################################################
+  '/export/api':
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml]: Correct the path as discussed during the code review

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml [hunk 9]
```diff
@@ -6400,6 +6400,90 @@ paths:
...
+            - YAML
+          required: true
+        - name: preserveStatus
+          in: query
+          description: |
+            Preserve API Status on export
+          required: false
+          type: boolean
+      tags:
+        - API (Individual)
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml]: APIs

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml (hunk 8 above)
Comment (malinthaprasan) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml]: /apis/export

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/components/APIDetailsTopMenu.jsx [hunk 10]
```diff
@@ -59,13 +66,70 @@ const styles = (theme) => ({
...
     const isVisibleInStore = ['PROTOTYPED', 'PUBLISHED'].includes(api.lifeCycleStatus);
+    /**
+ * The component for advanced endpoint configurations.
+ * @param {string} name The name of the
+ * @param {string} version Version of the API
+ * @param {string} provider Provider of the API
+ * @param {string} format Weather to recive files in YALM of JSON format
+ * @returns {zip} Zpi file containing the API directory.
+ */
+    async function exportAPI() {
```
Comment (malinthaprasan) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/components/APIDetailsTopMenu.jsx]: correct as discussed to remove the async

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ExportApiServiceImpl.java (hunk 7 above)
Comment (VimukthiMayadunne) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ExportApiServiceImpl.java]: pat changed now the function in APis api service impl

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ExportApiServiceImpl.java (hunk 5 above)
Comment (VimukthiMayadunne) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ExportApiServiceImpl.java]: Generated method name

**Pull request closed without merging**